### DISCUSSION TITLE
Reduce bundle size / improve tree shaking

### DISCUSSION THREAD

**Discussion by nandorojo** [software-mansion/react-native-reanimated#2843] [Ideas]
Update: I opened a PR here: https://github.com/software-mansion/react-native-reanimated/pull/3278

---

Reanimated's bundle size is currently massive, and it doesn't have any optimizations for tree shaking. As a result, by using Reanimated, my website is loading in a lot of code it doesn't actually use.

When I import it in a Next.js project, and run `ANALYZE=true yarn next build` using `@next/bundle-analyzer`, this is the output I see. 

![image](https://user-images.githubusercontent.com/13172299/149564148-8c859382-0e3e-48fc-b2ce-9a4dbd431047.png)

Reanimated and Gesture Handler add 60kb and 25kb gzipped (respectively).

`framer-motion` had a similar issue, but they optimized it very well. I'm wondering if there could be a similar solution for Reanimated.

What if we added an extra `tree` export, like:

```ts
import { createAnimatedComponent, withTiming, AnimatedView } from 'react-native-reanimated/tree'
```

That way, we could find a way to tree shake the library without causing a breaking change. 

The main export file could import from the `tree.ts` file, so that there isn't duplicate code.

**Comment by RazaShehryar** [comment]
really need this feature

**Comment by nandorojo** [comment]
I made a repo playground: https://github.com/nandorojo/reanimated-tree-shaking

**Comment by ponikar** [comment]
Just checking if we find a solution to reduce the bundle size. in our react native project the reanimated takes 510KB.
